proto/pulumi/provider.proto

```diff
@@ -152,6 +152,13 @@ service ResourceProvider {
     // `Diff` compares an existing ("old") set of resource properties with a new set of properties and computes the
     // difference (if any) between them. `Diff` should only be called with values that have at some point been validated
     // by a [](pulumirpc.ResourceProvider.Check) call.
+    //
+    // The provider's response must satisfy the following invariants:
+    //
+    // * All keys in Diff appear in DetailedDiff.
```

Comment:
This is not quite correct - `Diff` should only contain top-level keys.

Reply:
I don't see how those two statements contradict.

Reply:
All keys in DetailedDiff appear in Diff is incorrect since the keys in DetailedDiff are full paths, while Diff can only contain top-level keys

Reply:
Ah. So it's more like
- For each top-level key in `Diff` there is a matching property path in `DetailedDiff`.
- For each entry in `DetailedDiff`, its top-level property is in `Diff`.

Reply:
Yeah, maybe: `For each top-level key in Diff there is at least one matching property path in DetailedDiff.` to call out the fact that there might be multiple